tests/Signature/SignatureV4Test.php

```diff
@@ -277,6 +277,32 @@ public function testProvider()
     /**
      * @dataProvider testProvider
      */
+    public function testSignRequestUnsignedPayload($req, $sreq, $creq)
```

Comment:
Could you also add a black-box test that specifies a date, credentials, service, region, and expected signature with the unsigned option set?